### PR TITLE
Replace MagicSDK_Web3 with Web3.swift

### PR DESCRIPTION
**Pull request by Ariflo**

### CODE REVIEW

File: Sources/MagicSDK/Core/Provider/RpcProvider.swift [hunk 1]
```diff
@@ -83,8 +83,8 @@ public class RpcProvider: NetworkClient, Web3Provider {
 
 public typealias Web3ResponseCompletion<Result: Codable> = (_ resp: Web3Response<Result>) -> Void
 
-internal extension Web3BytesInitializable {
-    init(_ bytes: Web3BytesRepresentable) throws {
+extension BytesInitializable {
+    init(_ bytes: BytesConvertible ) throws {
```
Comment (Ariflo) [inline Sources/MagicSDK/Core/Provider/RpcProvider.swift:86]: LMK if there was some special reason for the name change here. 👍

Comment (Ethella) [reply Sources/MagicSDK/Core/Provider/RpcProvider.swift:86]: https://app.shortcut.com/magic-labs/story/36788/rename-bytesinitializable-extension
Yes, it's an old issue that collides with the other packages. Could you leave a note about it?

File: MagicSDK.podspec [hunk 2]
```diff
@@ -20,7 +20,7 @@ TODO: Add long description of the pod here.
 
   s.source_files = 'Sources/MagicSDK/**/*'
 
-  s.dependency 'MagicSDK-Web3', '~> 1.0'
+  s.dependency 'Web3', '~> 0.5.0'
   s.dependency 'MagicSDK-Web3/ContractABI', '~> 1.0'
```
Comment (Ethella) [inline MagicSDK.podspec:24]: These two dependencies can be updated to Web3/ContractABI and Web3/PromiseKit

File: Package.swift [hunk 3]
```diff
@@ -25,7 +25,7 @@ let package = Package(
         .target(
             name: "MagicSDK",
             dependencies: [
-                .product(name: "MagicSDK_Web3", package: "Web3.swift"),
+                .product(name: "Web3", package: "Web3.swift"),
                 .product(name: "Web3PromiseKit", package: "Web3.swift"),
```
Comment (Ethella) [inline Package.swift:29]: Same here

**Review by Ethella — APPROVED** [review]

**Pull request merged**